[docs] TLS routing FAQs (#30499)
gravitational/teleport: docs/pages/architecture/tls-routing.mdx

When forwarding the requests to the Proxy Service, the local proxy performs
necessary connection upgrades and sets the required SNI for the TLS handshake.

> `tsh kubectl` and `tsh kube exec` commands also starts a local proxy

Reviewer: ```suggestion
The `tsh kubectl` and `tsh kube exec` commands also start a local proxy
```

Teleport clients handle the situation similarly to when a layer 7 load balancer
is present, performing the connection upgrades.

Note that the load balancer must forward the TLS layer to the Teleport Proxy.
For example, an AWS Network Load Balancer (NLB) must use "TLS" protocol for the
target group.

### Will TLS routing work behind my reverse proxy?

TLS routing does support Teleport Proxy behind layer 7 reverse proxies. The
reverse proxy must support custom (connection
upgrades)[https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Upgrade)
and long-lived connections.

You can setup a Teleport Proxy behind the reverse proxy and run the following
command to test the connection:
```bash
$ curl -v https://<proxy.example.com>/webapi/connectionupgrade -H "Connection: Upgrade" -H "Upgrade: alpn-ping"
```

For a successful upgrade, the server should return "101 Switching Protocols"
with the requested upgrade type and you should also get some binary output in
about 30 seconds running the `curl` command:
```
< HTTP/1.1 101 Switching Protocols
< Connection: Upgrade
< Upgrade: alpn-ping
< X-Teleport-Upgrade: alpn-ping
<
Warning: Binary output can mess up your terminal.
```

> ### How to avoid `tsh proxy kube` for Kubernetes Access?

Reviewer: ```suggestion
### How can I access Kubernetes clusters without using `tsh proxy kube`?```

Local proxies are required when Teleport Proxy is behind layer 7 load balancers
or reverse proxies.

One alternative to `tsh proxy kube` is to use `tsh kubectl` which starts a
local proxy automatically when connection upgrades are required.

To completely avoid local proxies, either use layer 4 load balancers in TCP
mode instead of layer 7 load balancers, or configure your Teleport Proxy in

> `seperate` port mode and use a separate Kubernetes port that is not behind

Reviewer 2: ```suggestion
`separate ` port mode and use a separate Kubernetes port that is not behind
```
